datasets/bbox_classification_dataset.py

```diff
@@ -2,39 +2,63 @@
 import torch
 from ThermalClassifier.datasets.classes import BboxSample
 from pycocotools.coco import COCO
-import os.path as osp
-
+from pathlib import Path
 class BboxClassificationDataset(Dataset):
     def __init__(self,
                 root_dir: str,
                 class2idx: dict, 
                 annotation_file_name: str,
-                transforms = None) -> None:
-        
-        self.root_dir = root_dir
-        self.transforms = transforms
+                transforms = None,subsampling=1) -> None:
 
+        self.root_dir = Path(root_dir)
+        self.transforms = transforms
+        self.subsampling = subsampling
+        
         try:
-            data = COCO(osp.join(root_dir, annotation_file_name))
+            data = COCO(self.root_dir/annotation_file_name)
         except FileNotFoundError:
             raise Exception(f"{root_dir} does not have {annotation_file_name} !")
 
+        # This is a small patch to way⌊ the soi labels are currently structured, 
+        # there is need to append the img dir to the relative path in order to get the relevant frame
+        
+        if 'img_dir' in data.dataset['info']:
+            self.img_dir = data.dataset['info']['img_dir']
+        
+        else:
+            self.img_dir = None
+
+
         self.class_mapper = self.create_class_mapper(data.cats, class2idx)
 
         # only anns that has wanted classes !
         self.anns_dict = {ann_id: ann_dict for ann_id, ann_dict in data.anns.items() 
                                     if ann_dict['category_id'] in self.class_mapper}
         
+        # Subsample the video using the subsampling variable
+        self.anns_dict = {ann_id: ann_dict for ann_id, ann_dict in data.anns.items() 
+                                    if ann_dict['image_id'] % subsampling == 0}
```

Comment:
You are assuming that the image_id is the frame number. 
This assumption is good for soi but in other datasets, the image_id don't crossponding to the frame number. I think you should loop on anns with enumerate.